model/task/task.go

```diff
@@ -1173,10 +1195,10 @@ func (t *Task) MarkUnscheduled() error {
 }
 
 func (t *Task) MarkUnattainableDependency(dependency *Task, unattainable bool) error {
+	// check all dependencies in case of erroneous duplicate
```

Comment:
It looks to me like the database update will still be only for the first instance of the dependency in the array.

Reply:
Good catch!